docs/_extensions/simp_roles.py

```diff
@@ -0,0 +1,74 @@
+import re
+from docutils import nodes
+from docutils.parsers.rst import roles
+import sphinx.errors
+
+def setup(app):
+    app.add_role('pupmod',  pupmod_class())
+    app.add_role('package', auto_class('package'))
+    app.add_role('param',   auto_class('param', 'literal'))
+    app.add_role('jira',    auto_class('jira', 'inline', 'https://simp-project.atlassian.net/browse/%s'))
+    app.add_role('github',  github_class())
+
+# Ensure options dict contains a classes list with a 'simp-{role_name}' item
+def append_to_options_classes(options, role_name):
+  roles.set_classes(options)
+  if 'classes' not in options.keys():
+    options['classes'] = []
+  options['classes'].append( 'simp-%s' % role_name )
+  return(options)
+
+# Add a :<role_name>:`<text>` role processor function
+def auto_class(role_name,role_type='inline',pattern=False):
+  def role(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    options = append_to_options_classes(options, role_name)
+
+    if pattern:
+      if role_type == 'literal':
+        options = append_to_options_classes(options, 'literal')
+      # From: https://protips.readthedocs.io/link-roles.html
+      url = pattern % (text,)
+      node = nodes.reference(rawtext, text, refuri=url, **options)
+    elif role_type == 'literal':
+      node = nodes.literal(rawtext, text, **options)
+    else:
+      node = nodes.inline(rawtext, text, **options)
+
+    return [node], []
+  return role
+
+# Add a :pupmod:`<text>` role that auto-links "org-name" to the Puppet Forge
+def pupmod_class():
+  def role(name, rawtext, text, lineno, inliner, options={}, content=[]):
+    options = append_to_options_classes(options, 'pupmod')
+    forge_names = re.split("[-/]", text)
+
+    # Link to the module on Puppet Forge (if the text is in the right format)
+    if len(forge_names) == 2:
+      url = 'https://forge.puppet.com/%s/%s' % (forge_names[0], forge_names[1])
+      node = nodes.reference(rawtext, text, refuri=url, **options)
+      if re.search('/', text):
+        fixed_text = re.sub('/', '-', text)
+        print("WARNING: Use '-' to separate Puppet module names (.e.g., %s', not '%s')\n" % (fixed_text, text))
```

Comment:
I think that this warning is misplaced and we should always use the `author/module_name` format.

Reasons:

* There doesn't appear to be any definitive guidance on this and I think that it's easier to read as `author/module_name`
* The forge lists modules as `author/module_name`
* The dependencies list in all modules that I've generally seen list as `author/module_name`

Differences:

* The Puppetfile appears to like `author-module_name`
* The `name` entry in the metadata.json needs to be `author-module_name`

So, personally, I'd like to come down on the side of 'easier to understand' with `author/module_name`.

Reply:
Currently, the validation doesn't match the example.

Reply:
@trevor-vaughan :+1:  A few years ago, Puppet started preferring `org-name` over `org/name`, as seen in metadata.json, Puppet commands, and (I think) the Forge API.  However, the most relevant guidance I've found has been the [official Puppet documentation on module _metadata_][1]:

> "**Tip**: Although the Forge expects to receive modules named `username-module`, its web interface presents them as `username/module`. Always use the `username-module` style in your metadata files and when issuing commands.

I'll still be glad to remove the warning—that metadata guidance doesn't explicitly apply to documentation nomenclature, and if there is such guidance Puppet is obviously not in a hurry to apply it to their own docs.  I personally prefer `org/name` because it makes the namespacing obvious and is less easy to mistake for an RPM or program name.


[1]: https://puppet.com/docs/puppet/6.17/modules_publishing.html#naming-your-module

Reply:
Warning removed in latest commit; all valid delimiters are welcome in our docs!